records_mover/url/gcs/gcs_directory_url.py

```diff
@@ -38,7 +38,10 @@ def purge_directory(self) -> None:
             dir_loc.purge_directory()
         for file_loc in self.files_in_directory():
             file_loc.delete()
-        self.bucket_obj.blob(self.blob).delete()
+        # Like S3, directories are just a convenient fiction in GCS;
+        # once all contents are gone, they disappear from the file
+        # listing, since they never really existed, so we don't need
+        # a final delete on the directory.
```

Comment:
Looks like there was a bug in #61 that did a delete that shouldn't have been there.